src/payment/checkout/payment-form/StripePaymentForm.jsx

```diff
@@ -112,94 +120,47 @@ function StripePaymentForm({
-          skus,
-        });
-        await getAuthenticatedHttpClient()
-          .post(
-            `${process.env.STRIPE_RESPONSE_URL}`,
-            postData,
-            {
-              headers: { 'Content-Type': 'application/x-www-form-urlencoded' },
+    try {
+      const result = await stripe.updatePaymentIntent({
+        elements,
+        params: {
+          payment_method_data: {
+            billing_details: {
+              address: {
+                city,
+                country,
+                line1: address,
+                line2: unit || '',
+                postal_code: postalCode || '',
+                state: state || '',
+              },
+              email: context.authenticatedUser.email,
+              name: `${firstName} ${lastName}`,
             },
-          )
-          .then(response => {
-            setLocation(response.data.receipt_page_url);
-          })
-          .catch(error => {
-            const errorData = error.response ? error.response.data : null;
-            if (errorData && error.response.data.sdn_check_failure) {
-              // SDN failure: redirect to Ecommerce SDN error page.
-              setLocation(`${getConfig().ECOMMERCE_BASE_URL}/payment/sdn/failure/`);
-            } else if (errorData && errorData.sku_error) {
-              skuErrorDispatcher();
-            } else if (errorData && errorData.user_message) {
-              // Stripe error: tell user.
-              issueErrorDispatcher();
-            } else {
-              // Unknown error: log and tell user.
-              logError(error, {
-                messagePrefix: 'Stripe Submit Error',
-                paymentMethod: 'Stripe',
-                paymentErrorType: 'Submit Error',
-              });
-              issueErrorDispatcher();
-            }
-            setIsLoading(false);
-          });
-      }
-    };
-
-    const result = await stripe.updatePaymentIntent({
-      elements,
-      params: {
-        payment_method_data: {
-          billing_details: {
-            address: {
-              city,
-              country,
-              line1: address,
-              line2: unit || '',
-              postal_code: postalCode || '',
-              state: state || '',
+            metadata: {
+              organization,
+              purchased_for_organization: purchasedForOrganization,
             },
-            email: context.authenticatedUser.email,
-            name: `${firstName} ${lastName}`,
-          },
-          metadata: {
-            organization,
-            purchasedForOrganization,
           },
         },
-      },
-    });
-    stripePaymentMethodHandler(result);
+      });
+      onSubmitPayment({
+        paymentIntentId: result.paymentIntent.id,
+        skus,
+      });
+    } catch (error) {
+      // Show updatePaymentIntent error by the Stripe billing form fields
+      if (error.type === 'card_error' || error.type === 'validation_error') {
```

Comment:
I've been wanting to remove them but I thought we discussed to keep it? I could be wrong with my memory. This could be just remnant of the initial Stripe implementation (following the docs). There is a difference in the way the `PaymentElement` does validation and how we do it with the `CardHolderInformation`, in that Stripe displays the validation error message by the field before the submit is fired, while we display the validation error only after.

This error message is displayed after the submit, but I don't see why we need it tbh.

Reply:
Update: planning to merge this before modifying for QA in Stage, will follow up in the next PR if necessary.